spec/models/user_spec.rb

```diff
@@ -114,10 +114,22 @@
         before { user.require_password_confirmation = true }
 
         context 'when the current password is correct' do
-          before { user.update_attributes(password: 'newpass123!', current_password: 'goodpass1!') }
 
-          it 'updates the password' do
-            expect(user.valid_password?('newpass123!')).to be true
+          context 'when the new password is different from the current password' do
+            before { user.update_attributes(password: 'newpass123!', current_password: 'goodpass1!') }
```

Comment:
I forgot to mention that you can call `update` instead of `update_attributes`. They are equivalent, but Rails 4 lets us use `update`. And again, bang methods are preferred in tests, so you can use `update!` here and in the following spec.